pyproject.toml

```diff
@@ -19,15 +19,16 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.dependencies]
 python = ">=3.10,<3.12" # updated for llama-index dependency
-openai = "0.28"
+openai = "^1.0"
 termcolor = "2.3.0"
 typer = ">=0.3.2"
 rudder-sdk-python = ">=2.0.2"
 dataclasses-json = "0.5.7"
 tiktoken = ">=0.0.4"
 tabulate = "0.9.0"
 python-dotenv = ">=0.21.0"
-langchain = ">=0.0.335"
+langchain = "^0.1"
+langchain_openai = "^0.0.2"
```

Comment:
My bad before. Given it's still on a pre-`0.1` release, adding a `^` will basically be the same as `==0.0.2`.

Given this is an extra for langchain, we can probably rely on it being correctly resolved with its openai and langchain version requirements.

```suggestion
langchain_openai = "*"
```